### PR TITLE
[RFC] BCDA-1251: Updates to "security implications" in pull request template 

### PR DESCRIPTION
**Pull request by rnagle**
### Proposed changes

This change includes:

- Updates for our pull request template, primarily around the "security implications" section.
- Minor formatting changes, clean up through the pull request template.

### Change Details

Rationale: in making this change, we're hoping to refine our process for identifying and addressing security concerns in our work and make use of pull requests as a channel for communicating those concerns to the rest of the team and project stakeholders.

Since reviewers typically won't see them, I am highlighting guiding questions embedded in the PR template in **bold** below:

**Does the change deal with PII/PHI at all? What should reviewers look for in terms of security concerns?**

1. **YES or NO**: This change adds new software dependencies

**If yes, list the new dependencies and briefly note any relevant security impacts**

2. **YES or NO**: This change alters security controls or supporting software

**If yes, what security controls or supporting software are affected?**

3. **YES or NO**: This change introduces storage or transmission of new data

**If yes, what new data are we storing or transmitting? Is the data considered PII/PHI?**

4. **YES or NO**: A security checklist has been completed for this change

**If yes, provide a link to the security checklist in Confluence here.**

### Security Implications

1. **NO**: This change adds new software dependencies
2. **NO**: This change alters security controls or supporting software
3. **NO**: This change introduces storage or transmission of new data
4. **NO**: A security checklist has been completed for this change

These changes are not related to a security concern in particular, but generally to how the team thinks about security processes and workflows.

### Feedback Requested

The format of the security implications section has changed/evolved over the course of several conversations among the team. Please add a comment on the general structure/format (i.e., the relative "usability" of the template) as well as any feedback you have re: does this help accomplish the goal identified in the "change details" section above. If it does not, how can it be better?

Some examples of PRs that have, in my opinion, used some variation of this template effectively:

- https://github.com/CMSgov/bcda-ops/pull/262
- https://github.com/CMSgov/bcda-app/pull/341
- https://github.com/CMSgov/bcda-app/pull/336

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -10,27 +10,45 @@
...
 <!-- Replace xxx with the JIRA ticket number: -->
+
 ### Fixes [BCDA-xxx](https://jira.cms.gov/browse/BCDA-xxx)
 
-<!-- describe the problem being solved here -->
+<!-- Describe the problem being solved here: -->
 
-### Proposed changes:
-<!-- List of changes with bullet points here: -->
+### Proposed changes
```
Comment (embh) [inline .github/pull_request_template.md]:
```suggestion
### Proposed Changes
```
While we're here, can this heading be capitalized the same way as the others? 🙏

**Comment by dhgreene** [comment]
I'd suggest wording that doesn't imply a yes or a no.  What about something like:

### Security Implications
**NO**: new software dependencies
**NO**: security controls or supporting software altered
**NO**: new data stored or transmitted 
**NO**: security checklist is completed for this change

**Comment by rnagle** [comment]
I think I see what you mean. To make sure I understand: you're suggesting a change to wording to remove a bias towards yes or no. The PR author still provides a "yes" or "no" qualifier/answer for each item. Did I get that right?

> I'd suggest wording that doesn't imply a yes or a no. What about something like:
> 
> ### Security Implications
> **NO**: new software dependencies
> **NO**: security controls or supporting software altered
> **NO**: new data stored or transmitted
> **NO**: security checklist is completed for this change

**Comment by StewGoin** [comment]
> I think I see what you mean. To make sure I understand: you're suggesting a change to wording to remove a bias towards yes or no. The PR author still provides a "yes" or "no" qualifier/answer for each item. Did I get that right?

I do like something that helps push the point of "we need to make sure security implications are apparent and agreed on, so there is no correct answer except the correct answer for the PR. NO is not better than YES, etc..."

**Comment by dhgreene** [comment]
> I think I see what you mean. To make sure I understand: you're suggesting a change to wording to remove a bias towards yes or no. The PR author still provides a "yes" or "no" qualifier/answer for each item. Did I get that right?

@rnagle yes.  For instance, there's some cognitive dissonance reading, "No: a security checklist has been completed for this change.

**Comment by rnagle** [comment]
Rather than including "YES or NO" in the template, I'm thinking it makes sense to just use a markdown checklist:

```
- [ ] new software dependencies
- [ ] security controls or supporting software altered
- [ ] new data stored or transmitted
- [ ] security checklist is completed for this change
```

Rendered:

- [ ] new software dependencies
- [ ] security controls or supporting software altered
- [ ] new data stored or transmitted
- [ ] security checklist is completed for this change

One caveat: Github treats checklists within PRs as "task lists" and will display progress for the PR. For example, see BCDA-1547 in the screenshot below:

image

It's unfortunate that there's no way to override this behavior, but I think the literal checklist a nice, if minor, improvement to usability. If there are no objections, I will make adjustments to the template along those lines.

**Comment by embh** [comment]
I used checkboxes on my latest PR, if anyone wants to see it in action: https://github.com/CMSgov/bcda-app/pull/347

**Comment by rnagle** [comment]
@StewGoin how about final checklist item along the lines:

```
- [ ] requires more information or team discussion to evaluate security implications
```

With a commented prompt like:

```
<!-- Use this to indicate you're unsure how this change may impact system security 
and would like to solicit the team's feedback. Optionally, provide background information
regarding your questions and concerns. -->
```

**Pull request merged**